src/clj_cbor/decoder.clj

```diff
@@ -0,0 +1,45 @@
+(ns clj-cbor.decoder
```

Comment:
I'm not thrilled about separating out this namespace since the main `codec` namespace has some more decoding logic which is more CBOR-specific, as well as the fact that for compatibility, the namespaced keyword needs to get used here for duplicate key error.

The alternative was having the jump table need to get passed in a "decode-value" function, but it seemed odd to do that since the jump table needs to be aware of the input buffer and more low-level things.

Reply:
Yeah this feels a bit forced - given the size of the code (~40 LOC for `decoder`, ~100 for `jump`) maybe it makes sense to just have it all in `codec` and avoid the extra separation? You could declare the jump entries in line with the literate sections for each major type, which could improve readability.